### ISSUE TITLE
issue with --ws-max-connections on http requests

### ISSUE THREAD

**Issue by ehsanhajian** [paritytech/substrate#13561]
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Description of bug

following https://substrate.stackexchange.com/questions/7485/parachain-rpc-errors-since-polkadot-0-9-36-client-upgrade post. I believe the issue is related to  `--ws-max-connections` . it only works on WS connection, not on HTTP requests. 
<img width="1075" alt="Screenshot 1401-12-17 at 18 24 37" src="https://user-images.githubusercontent.com/18378194/223741212-8744a3b4-5035-4068-aa88-d95d2fac4374.png">

<img width="333" alt="Screenshot 1401-12-17 at 18 17 25" src="https://user-images.githubusercontent.com/18378194/223741241-f84ca5d4-7229-4a9f-b495-ae6458a45552.png">

<img width="952" alt="Screenshot 1401-12-17 at 15 56 33" src="https://user-images.githubusercontent.com/18378194/223741303-b000ca4f-e65a-4c9a-b798-326c542e449a.png">

<img width="925" alt="Screenshot 1401-12-17 at 15 54 57" src="https://user-images.githubusercontent.com/18378194/223741367-f4b24783-e734-4e6e-b7ab-c4a14098ae1e.png">

As you can see in attached images I have a node for both WS and HTTP requests. I set `--ws-max-connections 50000` but it only works on WS requests not on HTTP requests. when I sent more than 100 http requests it started to to generate `"Transport endpoint is not connected"` error/message. It has nothing with SSL/TLS. I tested on both http and https protocols.

### Steps to reproduce

I use https://github.com/shazow/ethspam and https://github.com/INFURA/versus to generate valid jsonrpc queries. the command that I use 
```
ethspam --rpc http://endpointurl | versus --concurrency=100 --stop-after=10000 http://endpointurl
```

**Comment by bkchr** [comment]
CC @niklasad1

**Comment by niklasad1** [comment]
@ehsanhajian 

You are doing the call against the old HTTP server right i.e, port `--rpc-port`, correct? It will use the default value by jsonrpsee which causes that to be 100 which was a breaking change I realized now.

The server running on `--rpc-ws-port` will use this `--ws-max-connections`  so you could actually try your script against that one, it should work both on HTTP and WebSocket.

Thus, if we continue and merge https://github.com/paritytech/substrate/pull/13384 this will be fixed. Otherwise,  I suppose we need to expose another CLI flag for the "HTTP server only" or just hardcode it some huge value.

**Comment by niklasad1** [comment]
sorry my bad, it is `--ws-port` but which port are you doing the HTTP requests to?

**Comment by niklasad1** [comment]
As I tried to explain earlier the 9933 endpoint has the number of connection hardcoded to 100 but 9944 support both HTTP and WS.

Can you double check doing HTTP request against 9944 as well? :pray:

**Comment by ehsanhajian** [comment]
yes. That's correct. 9944 can support both `http` and `ws` requests.

**Comment by ehsanhajian** [comment]
but we still have this message 
```
2023-03-09 16:20:07 HTTP serve connection failed hyper::Error(Io, Os { code: 104, kind: ConnectionReset, message: "Connection reset by peer" })
```
is it only information log or an error?

**Comment by niklasad1** [comment]
I would not regard it as an error, the remote peer/client closed the connection

I think we can make this log `debug` instead as it causes more confusion than good.
I expect this log only be visible if something goes wrong by reading the socket

**Comment by ehsanhajian** [comment]
well even in debug mode I couldn't find anything related to it (at least for me) we just have many connectivity error to peers node. it seems because port 30333is not reachable. I'm not sure if it's relevant to `"Transport endpoint is not connected"` or not.
```
Mar 13 07:17:50 poc-0 astar-collator[1441909]: 2023-03-13 07:17:50.112 DEBUG tokio-runtime-worker libp2p_swarm: [Parachain] Connection attempt to PeerId("12D3KooWMJTknQ69Tu46pBygus1YhZhA9X7bQPBpY8zpRmSYuJtT") failed with Transport([("/ip4/18.181.186.198/tcp/30333/ws/p2p/12D3KooWMJTknQ69Tu46pBygus1YhZhA9X7bQPBpY8zpRmSYuJtT", Other(Custom { kind: Other, error: Timeout }))]).
Mar 13 07:17:54 poc-0 astar-collator[1441909]: 2023-03-13 07:17:54.696  WARN tokio-runtime-worker jsonrpsee_server::server: HTTP serve connection failed hyper::Error(Shutdown, Os { code: 107, kind: NotConnected, message: "Transport endpoint is not connected" })
Mar 13 07:17:54 poc-0 astar-collator[1441909]: 2023-03-13 07:17:54.696  WARN tokio-runtime-worker jsonrpsee_server::server: HTTP serve connection failed hyper::Error(Shutdown, Os { code: 107, kind: NotConnected, message: "Transport endpoint is not connected" })
Mar 13 07:18:49 poc-0 astar-collator[1441909]: 2023-03-13 07:18:49.840 DEBUG tokio-runtime-worker libp2p_swarm: [Parachain] Connection attempt to PeerId("12D3KooWNYkYSHkjpGZi7RemovedByMe") failed with Transport([("/ip4/52.155.117.200/tcp/30333/ws/p2p/12D3KooWNYkYSHkjpGZi7RemovedByMe", Other(Custom { kind: Other, error: Timeout }))]).
```

**Comment by niklasad1** [comment]
libp2p and jsonrpc errors are not related in general, so for libp2p errors please open another issue.

As I said before just ignore this JSON-RPC server warning, it just that the client is not connected anymore.

**Comment by bLd75** [comment]
Thanks @niklasad1 for insights, we didn't get that both ports were merged behind `--ws-port` actually, it would be great if that was specified in docs and eventually merged behind `--rpc-port` for which terminology makes more sense for an http/ws server.

**Comment by niklasad1** [comment]
> By the way, would it be possible to set the message "Accepting new connection" in debug for HTTP server? It makes the node logs very large on an http server and doesn't really makes sense since connections don't stay open on it.

Yes

**Comment by niklasad1** [comment]
Closed by #13384 

Now, the `http server` has been removed an the `--rpc-max-connections` will configure it for a server which supports both http and websocket
